sql-plugin/src/main/330+/scala/com/nvidia/spark/rapids/shims/GpuIntervalUtils.scala

```diff
@@ -0,0 +1,556 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.nvidia.spark.rapids.shims
+
+import java.util.concurrent.TimeUnit.{DAYS, HOURS, MINUTES, SECONDS}
+
+import ai.rapids.cudf.{ColumnVector, DType, Scalar}
+import com.nvidia.spark.rapids.Arm
+
+import org.apache.spark.sql.types.{DayTimeIntervalType => DT}
+
+/**
+ * Parse DayTimeIntervalType string column to long column of micro seconds
+ * Spark DayTimeIntervalType type:
+ * https://spark.apache.org/docs/latest/sql-ref-datatypes.html
+ * Spark parse DayTime:
+ * https://github.com/apache/spark/blob/v3.2.1/
+ * sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/IntervalUtils.scala#L275
+ *
+ * DayTimeIntervalType have 10 sub types, 10 types and examples are as following:
+ *
+ * INTERVAL DAY: INTERVAL '100' DAY
```

Comment:
What happens if Spark sees `'100' DAY` on a type it expects to be INTERVAL DAY TO HOUR?

Reply:
`interval '100' DAY` does not match the regexp of `INTERVAL DAY TO HOUR` type and Spark return `NULL` when reading the CSV file. There are 10 regexps for each sub `DayTimeIntervalType` type.  Should explicitly specify the sub types in the schema when reading CSV files.

Reply:
Finally, `DayTimeIntervalGen` is simplified in the last code commit.